dependencydiff/mapping.go

```diff
@@ -78,12 +81,35 @@ var (
 		"composer": ecosystemPackagist,
 		"rubygems": ecosystemRubyGems,
 		"nuget":    ecosystemNuGet,
+		"actions":  ecosystemActions, /* created this entry for the "actions" not to fail in the mapping process. */
 	}
 )
 
+func mapDependencyEcosystemNaming(deps []dependency) error {
+	for i := range deps {
+		if deps[i].Ecosystem == nil {
```

Comment:
why does this not return an error too?

Reply:
hmm, since we allow the package ecosystem field to be null-able?

Reply:
ah, GH does not always populate it?
Let's add comments / notes to the code to explain.

Reply:
> ah, GH does not always populate it? Let's add comments / notes to the code to explain.

for the GitHub API as the data source, the `ecosystem` is a required field to return - but in our previous discussion we'd like to make this more generale and not limited by an specific API, so only we only made the `name` field a `string` (required).

i'll definitely add a comment here to explain